worker/src/ocall_bridge/bridge_api.rs

```diff
@@ -66,13 +65,37 @@ impl Bridge {
             .get_oc_api()
     }
 
+    pub fn get_ipfs_api() -> Arc<dyn IpfsBridge> {
+        debug!("Requesting IPFS OCall API instance");
+
+        COMPONENT_FACTORY
+            .read()
+            .as_ref()
+            .expect("Component factory has not been set. Use `initialize()`")
+            .get_ipfs_api()
+    }
+
     pub fn initialize(component_factory: Arc<dyn GetOCallBridgeComponents + Send + Sync>) {
         debug!("Initializing OCall bridge with component factory");
 
         *COMPONENT_FACTORY.write() = Some(component_factory);
     }
 }
 
+/// Factory trait (abstract factory) that creates instances
+/// of all the components of the OCall Bridge
+pub trait GetOCallBridgeComponents {
+    /// remote attestation OCall API
+    fn get_ra_api(&self) -> Arc<dyn RemoteAttestationBridge>;
+
+    /// on chain OCall API
+    fn get_oc_api(&self) -> Arc<dyn WorkerOnChainBridge>;
+
+    /// ipfs OCall API
+    fn get_ipfs_api(&self) -> Arc<dyn IpfsBridge>;
+}
```

Comment:
just moved here, out of the source file that also contains the implementation (in order to separate trait definition and implementation, since the implementation will remain in the worker, but the API will be part of the separate o-call bridge crate)